### PR TITLE
feat: change syntax for array length

### PR DESCRIPTION
**Pull request by mlutze**
Frees up records to have fields with the name "length"

### CODE REVIEW

File: main/src/library/Array.flix [hunk 1]
```diff
@@ -93,12 +93,12 @@ namespace Array {
...
     /// Returns `true` if the given array `a` is empty.
     ///
-    pub def isEmpty(a: Array[a, r]): Bool = a.length == 0
+    pub def isEmpty(a: Array[a, r]): Bool = length(a) == 0
 
     ///
     /// Returns the length of the array `a`.
     ///
-    pub def length(a: Array[a, r]): Int32 = a.length
+    pub def length(a: Array[a, r]): Int32 = $LENGTH$(a)
```
[the remark this answers is not in the document]
Comment (JonathanStarup) [reply main/src/library/Array.flix:101]: we could also use something in java https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Array.html#getLength-java.lang.Object-

[the remark this answers is not in the document]
Comment (mlutze) [reply main/src/library/Array.flix:101]: Isn't there overhead for making a function call though?

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/library/Array.flix:101]: Maybe yes, maybe no. I suggest we keep it in the compiler, since it does seem like an intrinsic.

**Comment by mlutze** [comment]
@jaschdoc since you are already working on things related to this, do you want to take over this PR?

**Comment by jaschdoc** [comment]
Sounds good

**Comment by jaschdoc** [comment]
Any idea why this is failing?

**Comment by mlutze** [comment]
> Any idea why this is failing?

Looks like most failures are from `Array.length` not being in the `MinLib`/`NixLib`.

**Comment by magnus-madsen** [comment]
Maybe @mlutze can finish this one and then @jaschdoc can take over?

**Comment by mlutze** [comment]
> Maybe @mlutze can finish this one and then @jaschdoc can take over (on other array related stuff)?

Fair enough.

**Pull request merged**